addons/mail/static/src/discuss/core/web/channel_member_list_patch.js

```diff
@@ -0,0 +1,26 @@
+import { ChannelMemberList } from "@mail/discuss/core/common/channel_member_list";
+import { AvatarCardPopover } from "@mail/discuss/web/avatar_card/avatar_card_popover";
+
+import { usePopover } from "@web/core/popover/popover_hook";
+import { patch } from "@web/core/utils/patch";
+
+patch(ChannelMemberList.prototype, {
+    setup() {
+        super.setup(...arguments);
+        this.avatarCard = usePopover(AvatarCardPopover, {
+            position: "right",
+        });
+    },
+    onClickAvatar(ev, member) {
+        if (!this.canOpenChatWith(member)) {
+            return;
+        }
+        const target = ev.currentTarget;
+        if (!this.avatarCard.isOpen) {
+            this.avatarCard.open(target, {
+                id: member.persona.userId,
```

Comment:
Are we sure we always have the `userId` available?
What happens if there is no user? (eg. became archived after joining the channel)

Reply:
as far as I tested, if there is no user, then it is not longer in the channel member list. if it is, then we have the record locally and the userId is always there. and the data is fetched every time the popover is opened so it stays synced with the server.